### PR TITLE
Trait greenlets

### PR DESCRIPTION
**Pull request by itziakos**
This PR adds support for traits change notification using greenlets and the gevent library.

Please note that it requires gevent version 1.0 and above

### CODE REVIEW

**Comment by dpinte** [comment]
@itziakos there are some typo's to fix.

File: traits/trait_notifiers.py [hunk 1]
```diff
@@ -629,3 +629,23 @@ class NewTraitChangeNotifyWrapper ( TraitChangeNotifyWrapper ):
...
+#-------------------------------------------------------------------------------
+#  'NewTraitChangeNotifyWrapper' class:
+#-------------------------------------------------------------------------------
+
+class GEventTraitChangeNotifyWrapper(TraitChangeNotifyWrapper):
+    """ Dynamic change notify wrapper, dispatching on a gevent event loop.
+
+    This class is in charge to dispatch trait change events to a dynamic
+    listener, typically created using the `on_trait_change` method and the
+    `dispatch` parameter set to 'new'.
```
Comment (itziakos) [inline traits/trait_notifiers.py]: `new` -> `gevent`

File: traits/tests/test_gevent_notifiers.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        for i in range(10):
+            obj.foo = float(i)
+            obj.boo = float(i)
+        gevent.wait()
+
+        # Each event should be a separate greenlet
+        self.assertEqual(len(self.greenlets), 18)
+
+        # Event should arrive out of order when some of the listeners
+        # release the thread.
```
Comment (itziakos) [inline traits/tests/test_gevent_notifiers.py]: `thread` -> `greenlet thread`

**Comment by itziakos** [comment]
This PR was a toy example to see how to add new notification methods/ I can fix this PR if there is any interest in having it

**Comment by mdickinson** [comment]
As discussed offline, I'm closing this PR as out of date. Nevertheless, this is an interesting proof of concept that we may want to build on in the future, and I've tagged the HEAD of the corresponding branch as [experimental/greenlets-dispatch](https://github.com/enthought/traits/tree/experimental/greenlets-dispatch). 

If we revisit this, we'd likely want to look at integration with asyncio rather than greenlets, but the concept would be much the same.

**Comment by mdickinson** [comment]
Opened #860 to consider applying the ideas in this PR to asyncio.

**Pull request closed without merging**